src/UpdatedPackagesCollector.js

```diff
@@ -127,7 +127,19 @@ export default class UpdatedPackagesCollector {
 
   collectUpdates() {
     return this.packages.filter(pkg => {
-      return this.updatedPackages[pkg.name] || (this.flags.onlyExplicitUpdates ? false : this.dependents[pkg.name]) || this.flags.canary;
+      if (this.updatedPackages[pkg.name]) {
+        return true;
+      }
+
+      if (!this.flags.onlyExplicitUpdates && this.dependents[pkg.name]) {
+        return true;
+      }
+
+      if (this.flags.canary && !pkg.isPrivate()) {
+        return true;
```

Comment:
Yea, I actually just ran into that problem. I've been using my fork for the last few days and one of my dev modules was needlessly marked as private, thus leading to canary versions going out of sync and breaking bootstrap.